src/library_manager/lib_manager.c

```diff
@@ -590,6 +666,13 @@ static int lib_manager_store_library(struct lib_manager_dma_ext *dma_ext,
 		return ret;
 	}
 
+#if CONFIG_LIBRARY_AUTH_SUPPORT
+	ret = lib_manager_auth_proc((void *)library_base_address,
```

Comment:
do we really need 2 calls?

Reply:
We should inline comment why we call twice. Maybe we try different keys ?

Reply:
Yes. There are different phases of authentication. 
We need AUTH_PHASE_FIRST and AUTH_PHASE_LAST to be called.